apps/gateway/test/champions_test.exs

```diff
@@ -300,6 +301,114 @@ defmodule Gateway.Test.Champions do
+      fetch_last_message(socket_tester)
+
+      assert_receive %WebSocketResponse{response_type: {:afk_rewards, %AfkRewards{afk_rewards: initial_afk_rewards}}}
+
+      # Check that te initial afk reward rates are all 0
+      assert Enum.all?(user.afk_reward_rates, fn rate -> rate.rate == 0 end)
+
+      # Check that the initial afk rewards are all 0
+      assert Enum.all?(initial_afk_rewards, fn reward -> reward.amount == 0 end)
+
+      # Set up a powerful team to win a level with the user. That should increment the afk rewards rates
+      Enum.each(user.units, fn unit ->
+        GameBackend.Units.update_unit(unit, %{level: 9999})
+      end)
+
+      [campaign_progression | _] = user.campaign_progresses
+      level_id = campaign_progression.level_id
+      SocketTester.fight_level(socket_tester, user.id, level_id)
+      fetch_last_message(socket_tester)
+
+      assert_receive %WebSocketResponse{
+        response_type: {:battle_result, _ = battle_result}
+      }
+
+      assert battle_result.result == "win"
+
+      # Get advanced user
+      SocketTester.get_user(socket_tester, user.id)
+      fetch_last_message(socket_tester)
+      assert_receive %WebSocketResponse{response_type: {:user, %User{} = advanced_user}}
+
+      # Check that the gold and gems afk rewards rates are now greater than the initial rewards
+      gold_currency_id = Currencies.get_currency_by_name!("Gold").id
+      gems_currency_id = Currencies.get_currency_by_name!("Gems").id
+
+      assert Enum.any?(advanced_user.afk_reward_rates, fn rate ->
+               rate.currency_id == gold_currency_id && rate.rate > 0
+             end)
+
+      assert Enum.any?(advanced_user.afk_reward_rates, fn rate ->
+               rate.currency_id == gems_currency_id && rate.rate > 0
+             end)
+
+      # Wait for the afk rewards to increase
+      :timer.sleep(2000)
+
+      # Claim afk rewards
+      SocketTester.claim_afk_rewards(socket_tester, advanced_user.id)
+      fetch_last_message(socket_tester)
+      assert_receive %WebSocketResponse{response_type: {:user, %User{} = claimed_user}}
+
+      # Check that the user has received gold and gems
+      assert Enum.any?(claimed_user.currencies, fn currency ->
+               user_gold_currency = Enum.find(claimed_user.currencies, &(&1.currency.name == "Gold"))
+               initial_gold = user_initial_currencies |> Enum.find(&(&1.currency.name == "Gold"))
+
+               user_gold_currency.amount > initial_gold.amount
+             end)
+
+      assert Enum.any?(claimed_user.currencies, fn currency ->
+               user_gems_currency = Enum.find(claimed_user.currencies, &(&1.currency.name == "Gems"))
+               initial_gems = user_initial_currencies |> Enum.find(&(&1.currency.name == "Gems"))
+
+               user_gems_currency.amount > initial_gems.amount
+             end)
+
+      # TODO: check that the afk rewards rates have been reset after [CHoM-380] is solved
```

Comment:
Let's create an issue for this too! And link it here so the task isn't lost in the code